### DISCUSSION TITLE
Feedback on 3.16.0 Medical Changes

### DISCUSSION THREAD

**Discussion by LinkIsGrim** [acemod/ACE3#9432] [General]
3.16.0 introduces two changes to damage handling:
A) Armor Scaling (#9216). This system takes armor's `passThrough` (what you see in the Arsenal as "Ballistic Protection") into account to limit tankiness. The changes should increase survibility of vanilla's armored (CSAT and Spetsnaz) uniforms and helmets, and reduce protection of the vests designed for explosive resistance (the Carrier GL Rig, infamously). Should you find any combination of equipment that's still under or over-protective (in relation to vanilla), please comment here. Include classnames and RPT so we can figure out which mod it's coming from and which unit/combination of equipment specifically.

You can also reduce the effect of this change by tweaking the newly added "Armor Passthrough Effect" setting if you'd like to make comparisons or prefer the old system.

B) Crew Damage (#9246). Since the release of the Medical Rewrite, crew inside exploding vehicles have come out (almost, sometimes completely) unscathed. This was due to an oversight in how we intercepted the engine's damage events. We believe we've covered all cases now, but if you still find a case of crew not taking damage when they should, report it.

The numbers for B didn't go through a lot of testing, and were mostly pulled out of my own head with comparatives to damage that should've been sustained had the crew been dismounted. If you have feedback on the severity, amount, or type of wounds caused, please post it here as well.

**Comment by STyx2909** [comment]
Joining the .rpt of this latest test. Antistasi campaign, played locally.

In the last 10 minutes, I roll over an enemy with a BMP-2K. The guy then stand up and receive many HMG shots. Fall down. Stand up and finally a last shot incapacitate him.

Checked my ACE settings.
Everything as it should in this situation.

Here is a copy, of such, but not from ACE RC1 options (they are enabled in Medical section). *and I closed the game already*.

[ACE Antistasi Takistan Restoring Honor.txt](https://github.com/acemod/ACE3/files/12741991/ACE.Antistasi.Takistan.Restoring.Honor.txt)
[Arma3_x64_2023-09-27_18-09-16.txt](https://github.com/acemod/ACE3/files/12742002/Arma3_x64_2023-09-27_18-09-16.txt)

**Reply by LinkIsGrim** [reply]
I'll need either the unit's classname or a list of their equipment to reproduce this.

**Reply by STyx2909** [reply]
RHS, 3CB factions. Takistan Army West. Infantry.

![Arma3_x64 2023-09-28 12-02-17](https://github.com/acemod/ACE3/assets/76572473/c63f4dcd-6bd7-4bcc-9bd8-9f33d3af9833)
![Arma3_x64 2023-09-28 12-01-17](https://github.com/acemod/ACE3/assets/76572473/45f8cf2e-af91-42d1-8b1c-0317db24024f)
![Arma3_x64 2023-09-28 12-01-25](https://github.com/acemod/ACE3/assets/76572473/89c02c08-86b8-4571-9bdc-10005d12182c)

**Reply by LinkIsGrim** [reply]
Found an edge case where non-penetrating wounds wouldn't kill (that is intentional) but also couldn't produce knockouts at all (not intentional).

Increased pain from bullet-inflicted bruises to compensate (a rework of pain simulation would be a better solution. Out of scope for 3.16.0 though). Changes'll be in RC2 or release, depending on what else needs changing.

Thanks for testing.

**Comment by acknowledge14** [comment]
Hi, as AIs can be set to perform medical treatment only when carrying necessary items( #9362), could you please do a little modify, so that there is a new option(set to off by default, of course) to make "ace_common_fnc_replaceRegisteredItems" apply on AIs?

I know #9362 is made for people who are willing to set custom AI loadouts, but I modified a tiny addon that could make vanilla Med Kits convert more ACE medical items, which may do some benefit to AI medics.

@PabstMirror has listed some code:

["CAManBase", "init", {
    params ["_unit"];
    if (local _unit) then {
        [_unit] call ace_common_fnc_replaceRegisteredItems
    };
}] call CBA_fnc_addClassEventHandler;

**Reply by LinkIsGrim** [reply]
Implemented in #9458

**Comment by acknowledge14** [comment]
Hi, I found out sometimes the FAK AI conversion founction will not work.
Test only with lastest workshop ACE, CBA and NATO US Army Overhauled [RHS]
 https://steamcommunity.com/sharedfiles/filedetails/?id=1458833592
with #9362 and #9458 enabled.
I made a simple test mission(see below)

1 When start the mission, player's FAK will convert successfully ( I dont know if the image will be displayed correctly, but all 3 can be found in the upload zip below ).
![1](https://github.com/acemod/ACE3/assets/50673495/67c0f1a7-791f-4e33-814a-175776fa42f2)

2 then press M to open map, check AI teammate (No.6 autorifleman)'s inventory, you will find vanilla FAKs still there.
![2](https://github.com/acemod/ACE3/assets/50673495/8ba2d53f-8522-4fc4-81dd-6264aecfd946)

3 check another AI teammate (No.8 medic)'s backpack, the FAKs and Medikit are totally not converting to ace items.
![3](https://github.com/acemod/ACE3/assets/50673495/7a9bf42a-fd8f-4689-b8a0-75763901689c)


I also found NATO US Army Overhauled [RHS] mod uses a file called autogen.hpp to restructure NATO's equipments, some lines list here:
`    class B_Soldier_SL_F : B_Soldier_03_f_OCimport_02 {
        author = "jarrad96";
        scope = 2;
        scopeCurator = 2;
        displayName = "Squad Leader";
        side = 1;
        faction = "BLU_F";

        identityTypes[] = { "Head_NATO" , "LanguageENG_F" , "G_NATO_default" };


        class EventHandlers : EventHandlers {
            class CBA_Extended_EventHandlers : CBA_Extended_EventHandlers_base {};

            class ALiVE_orbatCreator {
                init = "if (local (_this select 0)) then {_onSpawn = {_this = _this select 0;sleep 0.2; _backpack = gettext(configfile >> 'cfgvehicles' >> (typeof _this) >> 'backpack'); waituntil {sleep 0.2; backpack _this == _backpack};_this setunitloadout [['rhs_weap_m4a1','rhsusf_acc_SFMB556','rhsusf_acc_anpeq15side_bk','rhsusf_acc_ACOG3',['rhs_mag_30Rnd_556x45_M855_Stanag',30],[],'rhsusf_acc_grip3'],[],['rhsusf_weap_m9','','','',['rhsusf_mag_15Rnd_9x19_JHP',15],[],''],['rhs_uniform_cu_ocp',[['FirstAidKit',4],['SmokeShell',2,1]]],['rhsusf_iotv_ocp_Squadleader',[['rhsusf_mag_15Rnd_9x19_JHP',2,15],['rhs_mag_30Rnd_556x45_M855_Stanag',10,30],['SmokeShellBlue',2,1],['rhs_mag_m67',2,1]]],[],'rhsusf_ach_helmet_ESS_ocp','',['Binocular','','','',[],[],''],['ItemMap','ItemGPS','ItemRadio','ItemCompass','ItemWatch','rhsusf_ANPVS_14']];reload _this};_this spawn _onSpawn;(_this select 0) addMPEventHandler ['MPRespawn', _onSpawn];};";
            };

        };

        // custom attributes (do not delete)
        ALiVE_orbatCreator_owned = 1;

    };`



test mission:
[ACE%20FAK%20CONVERSION%20FAILED%20ON%20AI%20(NATO%20OVERHAULED%20RHS%20US).VR.zip](https://github.com/acemod/ACE3/files/13680327/ACE.20FAK.20CONVERSION.20FAILED.20ON.20AI.20.NATO.20OVERHAULED.20RHS.20US.VR.zip)
NATO US Army Overhauled [RHS] template:
[NATORHSArmy.zip](https://github.com/acemod/ACE3/files/13680328/NATORHSArmy.zip)
related images:
[ACE FAK CONVERSION FAILED ON AI images.zip](https://github.com/acemod/ACE3/files/13680381/ACE.FAK.CONVERSION.FAILED.ON.AI.images.zip)

**Reply by LinkIsGrim** [reply]
ALiVE's orbat creator uses a really shitty method to modify unit loadouts and supporting it is unfeasible. You're going to have to live with it (or disable AI item requirements).
